internal/command/meta_config.go

```diff
@@ -129,18 +130,26 @@ func (m *Meta) rootModuleCall(rootDir string) (configs.StaticModuleCall, tfdiags
 		name := variable.Name
 		v, ok := variables[name]
 		if !ok {
-			// For now, we are simply failing when the user omits a required variable. This is due to complex interactions between variables in different code paths (apply existing plan for example)
-			// Ideally, we should be able to use something like backend_local.go:interactiveCollectVariables() in the future to allow users to provide values if input is enabled
-			// This is probably blocked by command package refactoring
 			if variable.Required() {
-				// Not specified on CLI or in var files, without a valid default.
-				return cty.NilVal, hcl.Diagnostics{&hcl.Diagnostic{
-					Severity: hcl.DiagError,
-					Summary:  "Variable not provided via -var, tfvars files, or env",
-					Subject:  variable.DeclRange.Ptr(),
-				}}
+				// User prompts are best efforts, so we accept the input here
+				// and rely on downstream checks to validate it
+				rawVal, err := m.getInput(context.Background(), variable)
+				if err != nil {
+					return cty.NilVal, hcl.Diagnostics{&hcl.Diagnostic{
+						Severity: hcl.DiagError,
+						Summary:  fmt.Sprintf("Failed to request input from user for variable var.%s", variable.Name),
+						Subject:  variable.DeclRange.Ptr(),
+					}}
+				}
+				v = unparsedVariableValueString{
+					str:        rawVal,
+					name:       name,
+					sourceType: tofu.ValueFromInput,
+				}
+				variables[name] = v
```

Comment:
If anyone ever changes the implementation of m.collectVariableValues, this could break.  We should instead move this into a function beside collectVariableValues that performs the injection.